x-pack/plugins/security_solution/docs/prebuilt_rules_customization_rfc.md

```diff
@@ -0,0 +1,2867 @@
+      <td>Yes</td>
+      <td>
+        <pre>
+          {
+            rule_source: {
+              type: 'external',
+              isCustomized: true,
+              ...
+            },
+            immutable: true,
+          }
+        </pre>
+      </td>
+    </tr>
+    <tr>
+      <td><i>Customized</i> <b>Prebuilt rule</b> (already migrated, no customizations after update)</td>
+      <td>
+        <pre> 
+          {
+            type: 'external',
+            isCustomized: true,
+              ...
+          }
+          </pre>
+      </td>
+      <td><pre>true</pre></td>
+      <td>No</td>
+      <td>
+        <pre>
+          {
+            rule_source: {
+              type: 'external',
+              isCustomized: false,
+              ...
+            },
+            immutable: true,
+          }
+        </pre>
+      </td>
+    </tr>
+    <tr>
+      <td>Invalid case: Migrating a migrated non-customized prebuilt rule, with customizations. <br><br> `immutable` should never be false if `rule_source.type' is 'external'. Migration should correct this inconsistency.</td>
+      <td>
+        <pre> 
+          {
+            type: 'external',
+            isCustomized: false,
+              ...
+          }
+          </pre>
+      </td>
+      <td><pre>false</pre></td>
+      <td>Yes</td>
+      <td>
+        <pre>
+          {
+            rule_source: {
+              type: 'external',
+              isCustomized: true,
+              ...
+            },
+            immutable: true,
+          }
+        </pre>
+      </td>
+    </tr>
+
+  </tbody>
+</table>
+
+As explained before, all these endpoints suffer from the tightly-coupled logic problem described in the section [`Problem with tightly coupled logic in our endpoints`](#problem-with-tightly-coupled-logic-in-our-endpoints).
+
+We should therefore create new CRUD methods as needed to uncouple them logically and cleanly apply migrations to them.
+
+---
+
+#### Bulk editing rules
+
+The endpoint **Bulk Actions** - `POST /rules/_bulk_action` has the same outputs from the migration logic as the endpoints listed in the above section, so it was included in the scenarios in the table above. 
+
+However, this endpoint has some specific details that should be mentioned.
+
+Firstly, when bulk editing rules, the migration should be carried out for the following use cases:
+  - Bulk adding or deleting index patterns
+  - Bulk adding or deleting tags
+  - Updating rule schedules
+  - Adding rules actions
+  - Duplicating a rule (with the migration applied only to the new rule)
+
+Out of the actions mentioned above, the only use cases that should possibly result in the migration having a result of `rule_source.isCustomized = true` are the first three:
+  - Bulk adding or deleting index patterns
+  - Bulk adding or deleting tags
+  - Updating rule schedules
+
+That means that updating a rule's actions or duplicating a rule should not be considered a customization of a prebuilt rule.
```

Comment:
Nothing to elaborate actually, just stating the obvious: since we are generating a custom rule, "customization" is not possible/doesn't make sense. 

I'll remove the reference to "duplicating a rule" to avoid confusion,